sdk/provisioning/Azure.Provisioning/src/Resource.cs

```diff
@@ -74,20 +72,26 @@ internal void AddDependency(Resource resource)
         /// <param name="createProperties">Lambda to create the ARM properties.</param>
         /// <exception cref="ArgumentNullException">If <paramref name="scope"/> is null.</exception>
         protected Resource(IConstruct scope, Resource? parent, string resourceName, ResourceType resourceType, string version, Func<string, object> createProperties)
+            : this(scope, parent, resourceName, resourceType, version, createProperties, false)
+        {
+        }
+
+        internal Resource(IConstruct scope, Resource? parent, string resourceName, ResourceType resourceType, string version, Func<string, object> createProperties, bool isExisting)
```

Comment:
I prefer `isExisting` not be on the parameter list and is just set in the ctor with `init`

Reply:
is existing needs to be available to the constructor to make it easier to bypass calling GetAzureName for existing resources.